decidim-proposals/app/commands/decidim/proposals/admin/publish_participatory_text.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Proposals
+    module Admin
+      # A command with all the business logic when an admin imports proposals from
+      # a participatory text.
+      class PublishParticipatoryText < Rectify::Command
+        # Public: Initializes the command.
+        #
+        # form - A form object with the params.
+        def initialize(form)
+          @form = form
+        end
+
+        # Executes the command. Broadcasts these events:
+        #
+        # - :ok when everything is valid.
+        # - :invalid if the form wasn't valid and we couldn't proceed.
+        #
+        # Returns nothing.
+        def call
+          transaction do
+            resort_proposals(form)
+            publish_drafts
+          end
+
+          broadcast(:ok)
+        end
+
+        private
+
+        attr_reader :form
+
+        def resort_proposals(form)
+          form.proposals.each do |prop_form|
+            proposal = Decidim::Proposals::Proposal.where(component: current_component).find(prop_form.id)
+            proposal.set_list_position(prop_form.position) if proposal.position != prop_form.position
+          end
+        end
+
+        def publish_drafts
+          Decidim::Proposals::Proposal.where(component: current_component).drafts.find_each do |proposal|
+            Rails.logger.warn("Unable to publish participatory text #{proposal.id} due to: #{proposal.errors.full_messages}") unless proposal.update(published_at: Time.current)
```

Comment:
Why log this? If it's important we should raise an error.

Reply:
Well, I preferred to be permissive and let some proposals to fail but allow the rest to load instead of failing all, then log it to be able to follow the problem.

Anyway it was a fast decision, I can change to make it abort all the transaction. What do you think is better?

Reply:
Just logging the error makes the error silent, and the user is not aware that some proposals failed to be published. 

I think we should either show these errors to the end user, or abort the whole transaction.

Reply:
I'm aborting the transaction and reporting the errors to the end user.

Reply:
Nice!